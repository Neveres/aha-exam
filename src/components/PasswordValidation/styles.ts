import { css } from '@emotion/react'

import { colors, fontSize, fontWeight, fontFamily } from 'src/GlobalCss'

export const validationsContainer = css`
  box-sizing: border-box;
  position: relative;
  margin-top: 20px;
  padding: 3px 12px 0 12px;
  width: 335px;
  height: 226px;
  font-family: ${fontFamily.ubuntu};
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  box-shadow: 4px 4px 20px rgba(0, 0, 0, 0.3);
  border-radius: 8px;
  background: #242424;
`

export const validationContainer = css`
  width: 311px;
  height: 40px;
  display: flex;
  align-items: center;

  svg {
    width: 24px;
    height: 24px;
  }

  span {
    color: ${colors.label};
    padding-left: 10px;
    font-style: normal;
    font-weight: ${fontWeight.lighter};
    font-size: ${fontSize.normal};
    line-height: 150%;
    letter-spacing: 0.25px;
  }
`
