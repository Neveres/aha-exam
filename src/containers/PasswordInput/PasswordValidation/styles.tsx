import { css } from '@emotion/react'

export const hintsContainer = css`
  margin-top: 20px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 18px 12px;
  position: relative;
  width: 335px;
  /* height: 226px; */
  background: #242424;
  box-shadow: 4px 4px 20px rgba(0, 0, 0, 0.3);
  border-radius: 8px;
`

export const hintContainer = css`
  padding-bottom: 10px;
  display: flex;
  align-items: center;
  width: 250px;

  span {
    color: #ffffff;
    padding-left: 12px;
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 150%;
    letter-spacing: 0.25px;
  }
`
