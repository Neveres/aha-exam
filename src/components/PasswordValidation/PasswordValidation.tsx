/** @jsxImportSource @emotion/react */
import React from 'react'
import CheckCircleIcon from '@mui/icons-material/CheckCircle'
import CheckCircleOutlineRoundedIcon from '@mui/icons-material/CheckCircleOutlineRounded'

import { hintsContainer, hintContainer } from './styles'

const HINTS = [
  'Have at least one uppercase letter',
  'Have at least one lowercase letter',
  'Have at least one number',
  'Have at least one special character (!@#$...etc)',
  'Longer than 8 characters',
]

export const getPrefixIcon = (isValid: boolean) => {
  return isValid ? (
    <CheckCircleIcon
      sx={{
        color: '#00D1FF',
      }}
    />
  ) : (
    <CheckCircleOutlineRoundedIcon
      sx={{
        color: '#565656',
      }}
    />
  )
}

interface PasswordValidationProps {
  validationResult: boolean[]
}

export const PasswordValidation: React.FC<PasswordValidationProps> = (
  props,
) => {
  const { validationResult } = props

  return (
    <div css={hintsContainer}>
      {HINTS.map((hint, index) => {
        return (
          <div css={hintContainer} key={hint}>
            {getPrefixIcon(validationResult[index])}
            <span>{hint}</span>
          </div>
        )
      })}
    </div>
  )
}
