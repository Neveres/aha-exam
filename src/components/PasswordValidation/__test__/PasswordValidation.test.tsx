import React from 'react'
import { create } from 'react-test-renderer'

import { PasswordValidation } from '../PasswordValidation'

describe('PasswordValidation', () => {
  test('should render well', () => {
    const defaultValidationResult = Array(5).fill(false)
    for (let i = 0, maxI = 5; i < maxI; i++) {
      const validationResult = [...defaultValidationResult]
      validationResult[i] = true

      expect(
        create(
          <PasswordValidation validationResult={validationResult} />,
        ).toJSON(),
      ).toMatchSnapshot()
    }
  })
})
