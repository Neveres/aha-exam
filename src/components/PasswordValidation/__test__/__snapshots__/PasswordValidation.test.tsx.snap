// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordValidation getPrefixIcon should return expected result corresponding to input 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hczinf-MuiSvgIcon-root"
  data-testid="CheckCircleIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
  />
</svg>
`;

exports[`PasswordValidation getPrefixIcon should return expected result corresponding to input 2`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
  data-testid="CheckCircleOutlineRoundedIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
  />
</svg>
`;

exports[`PasswordValidation should render well 1`] = `
<div
  className="css-45nixu-PasswordValidation"
>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hczinf-MuiSvgIcon-root"
      data-testid="CheckCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <span>
      Have at least one uppercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one lowercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one number
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one special character (!@#$...etc)
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Longer than 8 characters
    </span>
  </div>
</div>
`;

exports[`PasswordValidation should render well 2`] = `
<div
  className="css-45nixu-PasswordValidation"
>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one uppercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hczinf-MuiSvgIcon-root"
      data-testid="CheckCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <span>
      Have at least one lowercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one number
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one special character (!@#$...etc)
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Longer than 8 characters
    </span>
  </div>
</div>
`;

exports[`PasswordValidation should render well 3`] = `
<div
  className="css-45nixu-PasswordValidation"
>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one uppercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one lowercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hczinf-MuiSvgIcon-root"
      data-testid="CheckCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <span>
      Have at least one number
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one special character (!@#$...etc)
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Longer than 8 characters
    </span>
  </div>
</div>
`;

exports[`PasswordValidation should render well 4`] = `
<div
  className="css-45nixu-PasswordValidation"
>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one uppercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one lowercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one number
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hczinf-MuiSvgIcon-root"
      data-testid="CheckCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <span>
      Have at least one special character (!@#$...etc)
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Longer than 8 characters
    </span>
  </div>
</div>
`;

exports[`PasswordValidation should render well 5`] = `
<div
  className="css-45nixu-PasswordValidation"
>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one uppercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one lowercase letter
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one number
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14t8b42-MuiSvgIcon-root"
      data-testid="CheckCircleOutlineRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.88-11.71L10 14.17l-1.88-1.88a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.59 2.59c.39.39 1.02.39 1.41 0L17.3 9.7c.39-.39.39-1.02 0-1.41-.39-.39-1.03-.39-1.42 0z"
      />
    </svg>
    <span>
      Have at least one special character (!@#$...etc)
    </span>
  </div>
  <div
    className="css-184ivag-PasswordValidation"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hczinf-MuiSvgIcon-root"
      data-testid="CheckCircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <span>
      Longer than 8 characters
    </span>
  </div>
</div>
`;
